src/circuit.rs

```diff
@@ -993,6 +998,39 @@ impl CircuitBuilder {
         }
         (acc_lt, acc_gt)
     }
+
+    pub fn push_condswap(
+        &mut self,
+        s: GateIndex,
+        x: GateIndex,
+        y: GateIndex,
+    ) -> (GateIndex, GateIndex) {
+        if x == y {
+            return (x, y);
+        }
+        let x_xor_y = self.push_xor(x, y);
+        let swap = self.push_and(x_xor_y, s);
+        let x_swapped = self.push_xor(x, swap);
+        let y_swapped = self.push_xor(y, swap);
+        (x_swapped, y_swapped)
+    }
+
+    pub fn push_sorter(
```

Comment:
If I get it right this is what is shown in the paper in Figure 2c as 2Sorter. Looks good!